src/btree/bt_curprev.c

```diff
@@ -553,6 +554,17 @@ __cursor_row_prev(WT_CURSOR_BTREE *cbt, bool newpage, bool restart, size_t *skip
 restart_read_page:
         rip = &page->pg_row[cbt->slot];
         WT_RET(__cursor_row_slot_key_return(cbt, rip, &kpack, &kpack_used));
+        /*
+         * If the cursor has prefix search configured we can early exit here, given the supplied
+         * prefix doesn't match the key we have.
+         */
+        if (F_ISSET(&cbt->iface, WT_CURSTD_PREFIX_SEARCH) && prefix != NULL &&
+          !__wt_prefix_match(prefix, &cbt->iface.key)) {
+            /* It is not okay for the user to have a custom collator. */
+            WT_ASSERT(session, CUR2BT(cbt)->collator == NULL);
+            WT_STAT_CONN_DATA_INCR(session, cursor_search_near_prefix_fast_paths);
+            return (WT_NOTFOUND);
```

Comment:
I *think* prev should reset the cursor. It's not followed by a subsequent next inside the search_near code. Alternatively you could update the search_near code to do the cursor reset whenever WT_NOTFOUND is going to be returned - that's only safe if that operation is idempotent (i.e: doing it twice doesn't matter).

Reply:
I don't think it does as the bt_cursor.c code does the reset itself here https://github.com/wiredtiger/wiredtiger/blob/0088889f364506a5cf68049c9bdf986f2c188f34/src/btree/bt_cursor.c#L727